Added a function to truncate strings given a maximum number of characters with its test case

baimamboukar opened the pull request:
# Welcome to Dart community

### **Describe your change:**

* [✅ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?


### **Checklist:**
* [✅ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [✅ ] This pull request is all my own work -- I have not plagiarized.
* [ ✅] I know that pull requests will not be merged if they fail the automated tests.
* [✅ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

baimamboukar committed:
Adding truncate_string.dart

A dart extension used to truncate long strings to a desired number of characters.

baimamboukar committed:
Merge pull request #1 from baimamboukar/baimamboukar-string-truncate

Adding truncate_string.dart

baimamboukar committed:
Merge branch 'TheAlgorithms:master' into master

baimamboukar committed:
added truncate string test case

akashgk reviewed (changes requested):
Please add the tests to the same file instead of creating multiple files for the same.

baimamboukar committed:
added test case in same file

Instead of adding test case in separate file, I have added the test right in the file

baimamboukar commented:
> Please add the tests to the same file instead of creating multiple files for the same. 

Done ✔️
I added the tests in the same file

akashgk commented on strings/truncate_string.dart:
```diff
@@ -0,0 +1,20 @@
+}
+
+void main() {
+  test(
+    "Truncate a long string given the max number of words and or suffix",
+    () {
+      String stringToTruncate =
+          "The beautiful thing with learning is that no one can take it away from you";
+      expect(truncateString(stringToTruncate, maxNumberOfWords: 40),
+          equals("The beautiful thing with learning is..."));
+    },
+  );
```
Add more test cases. More Strings and edge cases.

baimamboukar replied:
Hello @akashgk! 
Added more test cases...
Thanks for reviewing

akashgk reviewed (changes requested):
LGTM, Please add more tests.

baimamboukar committed:
added more test cases

jerold commented on strings/truncate_string.dart:
```diff
@@ -0,0 +1,37 @@
+
+String truncateString(String string,
+    {int maxNumberOfWords, String suffix = '...'}) {
+  return (string.length - 3) < maxNumberOfWords
```
- Ternary checks the character length of the string, not the word count as input parameters suggest.
- subtracting 3 from `string.length` seems to try and account for the length of the suffix, but if a different suffix is supplied, it's length may not be 3.
- If the string's number of words is less than the maxNumberOfWords, you won't have performed a truncation, and will not need the suffix/replacement. ("..." at the end implies there is more to the string's content, but this would not be true if nothing is cut off the end)

Coded-t commented:
please add more test
